### PR TITLE
New mobile nav

### PR DESCRIPTION
**Pull request by erikphansen**
mobile-nav

## Description
This PR totally rebuilds the Profile 2.0 mobile sub-nav. Instead of clicking a button that reveals a slide-over menu, that is essentially identical to the non-mobile sub-nav, this PR introduces an all-in-one mobile sub-nav that has two states: closed and open. The menu opens in place positioned on a layer above the main page content.

## Testing done
Local

## Acceptance criteria
- [ ] It works.™
- [ ] It does what it's supposed to do.™

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by sanlouise — COMMENTED** [review]
image

Is this OK by design? When scrolling we see the background content behind the menu - it seems like adding a full-width white background to the menu would be better UI, but I know this is not really up to us

**Comment by sanlouise** [comment]
You mentioned that there is still some existing Redux logic that's now moot, can you create a ticket for a cleanup so that it doesn't yet forgotten (if you haven't already)

**Review by sanlouise — APPROVED** [review]
Great work! 👍

File: src/applications/personalization/profile-2/components/ProfileSubNav.jsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        );
+      })}
+    </ul>
+  );
+}
+
+const ProfileSubNav = ({ routes }) => {
+  return (
+    <nav
+      className="va-sidebarnav"
```
[the remark this answers is not in the document]
Comment (erikphansen) [reply src/applications/personalization/profile-2/components/ProfileSubNav.jsx:30]: oh good call. will do.

**Comment by erikphansen** [comment]
I made that [Redux cleanup ticket](https://github.com/department-of-veterans-affairs/va.gov-team/issues/11274)

File: src/applications/personalization/profile-2/components/ProfileMobileSubNav.jsx [hunk 2]
```diff
@@ -95,6 +95,8 @@ const ProfileMobileSubNav = ({
 
       if (justSwitchedToMobile) {
         setIsMobile(true);
+        const menuButtonHeight = getElementHeight(theMenu.current);
```
Comment (erikphansen) [inline src/applications/personalization/profile-2/components/ProfileMobileSubNav.jsx:98]: If the page loads when the page is not mobile-width, then the height of the placeholder would never get properly set and then switching to mobile-width would result in the menu appearing to sit directly over the profile section title.

File: src/applications/personalization/profile-2/components/ProfileMobileSubNav.jsx [hunk 3]
```diff
@@ -178,7 +180,6 @@ const ProfileMobileSubNav = ({
               ref={openMenuButton}
               className="open-menu"
               type="button"
-              aria-controls="va-profile-sidebar"
```
Comment (erikphansen) [inline src/applications/personalization/profile-2/components/ProfileMobileSubNav.jsx]: now that the mobile nav is an all-in-one component I don't think this attribute is needed at all.

**Pull request merged**